src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+            if (!EnAn_ChangeCutscene(this, csId)) {
+                break;
+            }
+            FALLTHROUGH;
+        case 0x2:
+        case 0x4:
+        case 0x6:
+        case 0x8:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)),
+                                      this->actor.child);
+            }
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+        case 0x5:
+        case 0x7:
+            Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)), &this->actor);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        case 0x9:
+            CutsceneManager_Stop(csId);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+// Only used if Player is using Kafei's Mask or if Human and Promised midnight meeting
+s32 EnAn_MsgEvent_MidnightMeeting(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+
+    if (this->msgEventState == 0) {
+        func_800B7298(play, &this->actor, PLAYER_CSACTION_WAIT);
+        play->nextEntrance = ENTRANCE(STOCK_POT_INN, 3);
+        gSaveContext.nextCutsceneIndex = 0;
+        play->transitionTrigger = TRANS_TRIGGER_START;
+        play->transitionType = TRANS_TYPE_FADE_BLACK;
+        gSaveContext.nextTransitionType = TRANS_TYPE_FADE_BLACK_SLOW;
+        this->msgEventState++;
+    }
+
+    return 0;
+}
+
+s32 EnAn_MsgEvent_Cooking(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+    s32 ret = false;
+
+    switch (this->msgEventState) {
+        case 0x0:
+            EnAn_ChangeAnim(this, play, ENAN_ANIM_TASTE);
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+                this->msgEventState++;
+                ret = true;
+            }
+            break;
+
+        case 0x2:
+            EnAn_ChangeAnim(this, play, ENAN_ANIM_TASTE_END);
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+s32 EnAn_MsgEvent_LaundryPool(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+    s32 ret = false;
+
+    switch (this->msgEventState) {
+        case 0x0:
+            if ((Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK) ||
+                CHECK_WEEKEVENTREG(WEEKEVENTREG_TALKED_ANJU_IN_LAUNDRY_POOL)) {
+                this->msgEventState++;
+                // fallthrough
+            } else {
+                ret = true;
+                this->stateFlags |= ENAN_STATE_DRAW_KAFEIS_MASK;
+                this->msgEventState++;
+                break;
+            }
+            FALLTHROUGH;
```

Comment:
This upper fall-through at the end of the if block, that seems redundant? The fallthrough token is already placed down below at the edge of the case block, while the if block does fall down into the rest of the case block, the proximity is so close it's in sight without scrolling or anything.

Reply:
It is indeed redundant. This was suggested to make super sure the fallthrough is up there since it is a bit hidden.
This scenario is annoying to deal with.

Even if it is near one to the other, I think for now it is better to keep both. We can revisit cases like this in the future when we have more examples